src/main/docs/guide/httpServer/customArgumentBinding.adoc

```diff
@@ -0,0 +1,59 @@
+
+== Custom argument binding
+
+Micronaut uses an api:http.bind.binders.ArgumentBinderRegistry[ArgumentBinderRegistry] to look up api:core.bind.ArgumentBinder[ArgumentBinder]s
+capable of binding to the arguments in the controller methods. The default implementation will look for an annotation on
+the argument that is, itself, annotated with `@Bindable`. If one exists the argument binder registry will search for an
+argument binder that supports that annotation.
+
+If no fitting annotation is found Micronaut will try to find an argument binder that supports the argument's type.
+
+An api:core.bind.ArgumentBinder[ArgumentBinder] can be registered as a bean. In this case it is only present when a
+api:context.BeanContext[BeanContext] is initialized.
+
+Binders are called by the same server logic that calls the api:io.micronaut.http.annotation.Controller[Controller]. Once
+the request is received and the argument is resolved in the api:http.bind.binders.ArgumentBinderRegistry[ArgumentBinderRegistry]
+api:io.micronaut.http.filter.HttpFilter[HttpFilter]s are called before returning results to the Controller. Your
```

Comment:
The section about filters needs more explanation. We should say that argument binders can be called multiple times to attempt to bind an argument. That should include saying that they are called before the body is read and before any filters are executed.